### DISCUSSION TITLE
issue: Controller doesn't respect other Controllers

### DISCUSSION THREAD

**Discussion by SasSam** [react-hook-form/react-hook-form#7499] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-issue-with-react-datepicker-hre59

### Steps to reproduce

1. Open modal by clicking on the "show modal" button
2. Click on the "Submit" button (all fields marked as invalid)
3. Select a date in the Main date (all date fields marked as valid)

### Expected behaviour

Facts: 

- The project is using `Controller`, `useFieldArray`, `useFormContext`, `FormProvider` and `useForm`
- The "First name", "Main date", "Note" and "Start date" are required
- Date fields are using `react-datepicker`, hence wrapped with `Controller`

Expected:

- If "Main date", "Start date" and "End date" are empty, on submit "Main date" and "Start date" should be marked as invalid. Also, if I fill in the "Main date", "Start date" should remain invalid.
- If "Main date" is empty, but "Start date" is filled, on submitting "Main date" should be invalid, but "Start date" should be marked as valid.

### Relevant log output

```shell
All output can find on the console as submit does console.log().
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/useform/formstate

> formState is wrapped with a Proxy to improve render performance and skip extra logic if specific state is not subscribed to. Therefore make sure you invoke or read it before a render in order to enable the state update.

This is most likely due to isValid is not subscribed.

**Reply by SasSam** [reply]
Hi Bill,

Thanks for your answer, but unfortunately, I don't think I get it. From my point of view, the main issue is that the other fields get invalid states based on another field's validation. How would that `isValid` help on this?

Please check the following example: I filled in the form like this:
![image](https://user-images.githubusercontent.com/496429/149094260-58e68057-b455-494a-91ff-cbd23cdf9abc.png)

And when I submit it, the error array shows the following:
![image](https://user-images.githubusercontent.com/496429/149094441-859d68f6-4b9a-424e-8727-62e044699d54.png)

Could you please give me more hints? I appreciate any help you can provide.

**Reply by bluebill1049** [reply]
for any dependant validation, you should use trigger, hook form focus on field level.

**Reply by SasSam** [reply]
> for any dependant validation, you should use trigger, hook form focus on field level.

There is no dependant validation. All fields are independent. The `test` array is generated by `useFieldArray`, but all fields have their own "required" rule.

**Reply by bluebill1049** [reply]
> If "Main date", "Start date" and "End date" are empty, on submit "Main date" and "Start date" should be marked as invalid. Also, if I fill in the "Main date", "Start date" should remain invalid.

why should they remain invalid? if you are already corrected them?
